File: etl/models/candidatura.py
from sqlalchemy import Column, BigInteger, String, Integer, ForeignKey, Numeric
from .base import Base

class Candidatura(Base):
    __tablename__ = "candidatura"

    sq_candidato = Column(BigInteger, primary_key=True)
    nr_cpf_candidato = Column(String(11), ForeignKey("candidato.nr_cpf_candidato"), nullable=False)
    cd_eleicao = Column(BigInteger, nullable=False)
    nr_partido = Column(Integer, ForeignKey("partido.nr_partido"))
    sq_coligacao = Column(BigInteger, ForeignKey("coligacao.sq_coligacao"))
    nm_urna_candidato = Column(String)
    cd_cargo = Column(Integer)
    ds_cargo = Column(String)
    sg_uf = Column(String(2))
    nr_candidato = Column(Integer)
    cd_situacao_candidatura = Column(Integer)
    ds_situacao_candidatura = Column(String)
    cd_ocupacao = Column(Integer)
    ds_ocupacao = Column(String)
    foto_url = Column(String)
    st_reeleicao = Column(String(1))
    vr_despesa_max_campanha = Column(Numeric(15, 2))

File: src/backend/VoteBem/Entities/Candidatura.cs
namespace VoteBem.Entities
{
    public class Candidatura
    {
        public long SqCandidato { get; private set; }
        public string NrCpfCandidato { get; private set; } = null!;
        public long CdEleicao { get; private set; }
        public int? NrPartido { get; private set; }
        public long? SqColigacao { get; private set; }
        public string? NmUrnaCandidato { get; private set; }
        public int? CdCargo { get; private set; }
        public string? DsCargo { get; private set; }
        public string? SgUf { get; private set; }
        public int? NrCandidato { get; private set; }
        public int? CdSituacaoCandidatura { get; private set; }
        public string? DsSituacaoCandidatura { get; private set; }
        public int? CdOcupacao { get; private set; }
        public string? DsOcupacao { get; private set; }
        public string? FotoUrl { get; private set; }
        public string? StReeleicao { get; private set; }
        public decimal? VrDespesaMaxCampanha { get; private set; }

        public Candidato Candidato { get; private set; } = null!;
        public Partido? Partido { get; private set; }
        public Coligacao? Coligacao { get; private set; }

        public ICollection<BemCandidato> BensCandidato { get; private set; } = [];
        public ICollection<RedeSocial> RedesSociais { get; private set; } = [];
        public ICollection<CertidaoCriminal> CertidoesCriminais { get; private set; } = [];
        public ICollection<DespesaCandidato> Despesas { get; private set; } = [];
        public ICollection<NotaFiscal> NotasFiscais { get; private set; } = [];
        public ICollection<ResultadoTurno> ResultadosTurno { get; private set; } = [];
        public ICollection<MotivoCassacao> MotivosCassacao { get; private set; } = [];
        public PropostaGoverno? PropostaGoverno { get; private set; }

        protected Candidatura() { }
    }
}
